### ISSUE TITLE
Contributors Guide

### ISSUE THREAD

**Issue by Cynfusion** [anvilresearch/connect-docs#12]
Seems like this would be a good first step as we are moving towards our own model of how we want to work.

Suggest starting with https://github.com/nodejs/io.js/blob/master/COLLABORATOR_GUIDE.md and 
https://github.com/nodejs/io.js/blob/master/CONTRIBUTING.md as guidelines, but with our own secret sauce and priorities for the project.

I'd put this as a pretty high priority for the docs group.

**Comment by christiansmith** [comment]
Agreed on priority. I'll post a few thoughts on content specific to this project here later today.

**Comment by christiansmith** [comment]
Git message [conventions](https://docs.google.com/document/d/1QrDFcIiPjSLDn3EL15IJygNPiHORgU1_OOAqWjiDU5Y/edit) we use to [generate the changelog](https://github.com/btford/grunt-conventional-changelog) need to be documented or referenced in the contributors guide.

**Comment by christiansmith** [comment]
Here are a few more topics that ought to be covered:

* code conventions
* pull requests – how to group/separate commits, etc
* code reviews
* testing – how to run tests, testing conventions, when tests are required, etc.
* licensing (i.e., don't copy/paste GPL'd code, etc)
* call for contributions – some folks may want to get involved but don't know where to start. It would be good to point out some low hanging fruit.

Some of this may overlap a bit with governance, the CLA issue, etc.

**Comment by bnb** [comment]
I can work on this - I'll send a PR in the next few days.

**Comment by christiansmith** [comment]
@bnb let's try to work in the open as much as possible. Once you start drafting this up, I think you'll find we still have a lot of questions to answer.

**Comment by Cynfusion** [comment]
As we move forward on this, I'd expect we would want to be sure and cover (among other things, such as the list above):
-How we work ( @christiansmith please weigh in here especially)
-What we value (openness, honesty, collaboration, clean code) ( @christiansmith among other things I am sure)
-Issues and Pull Requests
-Technical HowTo
-Issue Contributions
-Code Contributions
-Some kind of "Hot Top Ten" things people can jump in to easily

How are we coming, @bnb?

**Comment by vsimonian** [comment]
Is there an update on this issue?

**Comment by Cynfusion** [comment]
Yes, it's in my repository still. I suggest we meet w Christian this week and discuss and then merge it.

**Comment by vsimonian** [comment]
Sweet! I'm super excited.

**Comment by vsimonian** [comment]
PR #22

**Comment by vsimonian** [comment]
PR #22 has been manually merged by @christiansmith. Marking issue as needing review.
